src/Language/SystemVerilog/Parser/Parse.y

```diff
@@ -721,27 +721,42 @@ AssignOption :: { AssignOption }
 -- for ModuleItem, for now
 AssertionItem :: { AssertionItem }
   : ConcurrentAssertionItem { $1 }
+  | DeferredImmediateAssertionItem { $1 }
 
 -- for Stmt, for now
 ProceduralAssertionStatement :: { Assertion }
   : ConcurrentAssertionStatement { $1 }
-  | ImmediateAssertionStatement  { $1 }
+  | SimpleImmediateAssertionStatement  { $1 }
+
+ImmediateAssertionStatement :: { Assertion }
+  : SimpleImmediateAssertionStatement  { $1 }
+  | DeferredImmediateAssertionStatement { $1 }
+
+DeferredImmediateAssertionItem :: { AssertionItem }
+  : Identifier ":" DeferredImmediateAssertionStatement { ($1, $3) }
+  |                DeferredImmediateAssertionStatement { ("", $1) }  
 
 ConcurrentAssertionItem :: { AssertionItem }
   : Identifier ":" ConcurrentAssertionStatement { ($1, $3) }
   |                ConcurrentAssertionStatement { ("", $1) }
+
 ConcurrentAssertionStatement :: { Assertion }
   : "assert" "property" "(" PropertySpec ")" ActionBlock { Assert (Concurrent $4) $6 }
   | "assume" "property" "(" PropertySpec ")" ActionBlock { Assume (Concurrent $4) $6 }
   | "cover"  "property" "(" PropertySpec ")" Stmt        { Cover  (Concurrent $4) $6 }
 
-ImmediateAssertionStatement :: { Assertion }
+DeferredImmediateAssertionStatement :: {Assertion }
```

Comment:
minor nit: let's put a space in front of `Assertion` to match the rest of the type signatures